core/pom.xml

```diff
@@ -357,6 +357,34 @@
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-crypto</artifactId>
     </dependency>
+
+    <!--
+     Testing Hive reflection needs hive on the test classpath only.
```

Comment:
Is that possible to avoid adding these dependencies to core? Does it make sense to do the test in `org.apache.spark.sql.hive`? cc @vanzin @mridulm